### PR TITLE
Added most native Windows dependencies to the fetch script

### PR DESCRIPTION
**Pull request by Mailaender**
Further work upon https://github.com/OpenRA/OpenRA/issues/6863 to make it possible to package this on Linux again. Otherwise it will unnecessarily fail with:

```
Fetching SDL2 from nuget
./fetch-thirdparty-deps.sh: line 51: nuget: command not found
```

even though all CLI dependencies are resolvable.

### CODE REVIEW

**Comment by Phrohdoh** [comment]
I don't understand how we will build the Windows packages with these changes.

A bit off topic: Why do we && on commands that won't fail instead of pushd and popd?

**Comment by Mailaender** [comment]
> I don't understand how we will build the Windows packages with these changes.

`make windows-dependencies` but that will be changed by https://github.com/OpenRA/OpenRA/pull/7345 soon.

> A bit off topic: Why do we && on commands that won't fail instead of pushd and popd?

I am not the expert here. All I know is that Makefile syntax is not exactly like shell. Have a look at https://www.gnu.org/software/make/manual/make.html

**Comment by Phrohdoh** [comment]
Thanks!

File: thirdparty/fetch-thirdparty-deps-windows.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+# Die on any error for Travis CI to automatically retry:
```
Comment (pchote) [inline thirdparty/fetch-thirdparty-deps-windows.sh]: We should merge #7345 first, then this can be removed.

Comment (Mailaender) [inline thirdparty/fetch-thirdparty-deps-windows.sh]: This is still useful.

Comment (Mailaender) [inline thirdparty/fetch-thirdparty-deps-windows.sh]: Actually, the other way round. This should get merged first as https://github.com/OpenRA/OpenRA/pull/7345 has to adapt to the changes here.

File: thirdparty/fetch-thirdparty-deps.ps1 [hunk 2]
```diff
@@ -1,3 +1,5 @@
+New-Item -ItemType Directory -Force -Path windows
```
Comment (ScottNZ) [inline thirdparty/fetch-thirdparty-deps.ps1]: This should use the alias `mkdir`

File: thirdparty/fetch-thirdparty-deps.ps1 [hunk 3]
```diff
@@ -69,5 +71,29 @@ if (!(Test-Path "Mono.Nat.dll"))
...
+	echo "Fetching Lua 5.1 from NuGet."
+	./nuget.exe install lua51.redist -Version 5.1.5
+	cp lua51.redist.5.1.5/build/native/bin/Win32/v120/Release/lua5.1.dll ./windows/lua51.dll
+	rmdir lua51.redist.5.1.5 -Recurse
+}
+
+if (!(Test-Path "windows/zlib1.dll"))
+{
+	echo "Fetching ZLib from NuGet."
+	nuget install zlib.redist -Version 1.2.8.7
```
Comment (ScottNZ) [inline thirdparty/fetch-thirdparty-deps.ps1]: This should have `./nuget.exe`

File: thirdparty/fetch-thirdparty-deps.ps1 [hunk 4]
```diff
@@ -69,5 +71,29 @@ if (!(Test-Path "Mono.Nat.dll"))
...
+	cp zlib.redist.1.2.8.7/build/native/bin/v120/Win32/Release/dynamic/stdcall/zlib.dll windows/zlib1.dll
+	rmdir zlib.redist.1.2.8.7 -Recurse
+}
+
+if (!(Test-Path "windows/soft_oal.dll"))
+{
+	echo "Fetching ZLib from NuGet."
+	nuget install openal.redist -Version 2.0.7
```
Comment (ScottNZ) [inline thirdparty/fetch-thirdparty-deps.ps1]: This should have `./nuget.exe`

File: thirdparty/fetch-thirdparty-deps.ps1 (hunk 2 above)
Comment (Mailaender) [inline thirdparty/fetch-thirdparty-deps.ps1]: Can you give me the full line? I can't test myself for the correct syntax and AppVeyor is slow to respond.

**Comment by Mailaender** [comment]
Updated.

File: thirdparty/fetch-thirdparty-deps.ps1 (hunk 2 above)
Comment (ScottNZ) [inline thirdparty/fetch-thirdparty-deps.ps1]:
```
mkdir windows -Force >$null
```

Comment (Mailaender) [inline thirdparty/fetch-thirdparty-deps.ps1]: Thanks, updated.

**Comment by LipkeGu** [comment]
:+1:

**Comment by Mailaender** [comment]
Removed https://www.nuget.org/packages/openal.redist/ for now as we switched to the incompatible and unavailable http://kcat.strangesoft.net/openal.html#download which is not yet packaged.

**Comment by LipkeGu** [comment]
now it works :)

**Comment by LipkeGu** [comment]
:+1:

**Comment by ScottNZ** [comment]
:+1:

**Pull request merged**